src/main/java/co/cask/spanner/sink/SpannerSink.java

```diff
@@ -0,0 +1,104 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.spanner.sink;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.data.batch.Output;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.dataset.lib.KeyValue;
+import co.cask.cdap.etl.api.Emitter;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.batch.BatchRuntimeContext;
+import co.cask.cdap.etl.api.batch.BatchSinkContext;
+import co.cask.gcs.GCPUtil;
+import co.cask.hydrator.common.ReferenceBatchSink;
+import co.cask.hydrator.common.batch.sink.SinkOutputFormatProvider;
+import co.cask.spanner.SpannerConstants;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.NullWritable;
+
+
+/**
+ * This class extends {@link ReferenceBatchSink} to write to Google Cloud Spanner.
+ *
+ * Uses a {@link SpannerOutputFormat} and {@link SpannerOutputFormat.SpannerRecordWriter} to configure
+ * and write to spanner. The <code>prepareRun</code> method configures the job by extracting
+ * the user provided configuration and preparing it to be passed to {@link SpannerOutputFormat}.
+ *
+ */
+@Plugin(type = "batchsink")
+@Name(SpannerSink.NAME)
+@Description("Batch sink to write to Cloud Spanner. Cloud Spanner is a fully managed, mission-critical, " +
+  "relational database service that offers transactional consistency at global scale, schemas, " +
+  "SQL (ANSI 2011 with extensions), and automatic, synchronous replication for high availability.")
+public final class SpannerSink
+  extends ReferenceBatchSink<StructuredRecord, NullWritable, StructuredRecord> {
+  public static final String NAME = "Spanner";
+  private final SpannerSinkConfig config;
+
+  /**
+   * Initializes <code>SpannerSink</code>.
+   * @param config
+   */
+  public SpannerSink(SpannerSinkConfig config) {
+    super(config);
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    config.validate();
+    pipelineConfigurer.getStageConfigurer().setOutputSchema(config.getSchema());
+  }
+
+  @Override
+  public void prepareRun(BatchSinkContext context) throws Exception {
+    config.validate();
+    Configuration configuration = new Configuration();
+    String projectId = GCPUtil.getProjectId(config.project);
+    configuration.set(SpannerConstants.PROJECT_ID, projectId);
+    if (config.serviceFilePath != null) {
+      configuration.set(SpannerConstants.SERVICE_ACCOUNT_FILE_PATH, config.serviceFilePath);
+    }
+    configuration.set(SpannerConstants.INSTANCE_ID, config.instance);
```

Comment:
The convention in OutputFormats (InputFormats too) is to have a static method in the OutputFormat that takes a Configuration and other set of arguments. The static method is responsible for setting the values in the Configuration. That way, logic is encapsulated in that single class. 

I think it makes sense to follow that patten here and add a `SpannerOutputFormat.configure(Configuration conf, SpannerSinkConfig spannerConf)` method